app/src/main/java/protect/card_locker/importexport/CatimaImporter.java

```diff
@@ -95,8 +91,6 @@ public void importCSV(Context context, SQLiteDatabase database, InputStream inpu
             default:
                 throw new FormatException(String.format("No code to parse version %s", version));
         }
-
-        bufferedReader.close();
```

Comment:
Won't we leak the object if we don't close it? 1f76809300cc81e3db441ecd47de7c34f4073013 implies we do.

Reply:
Ah oops I only meant to conditionally disable the close

Reply:
Actually if I close it in importData instead, it works better. I had to remove it from importCSV so it doesn't close the zip input stream prematurely.

Reply:
Does closing an InputStream close all InputStreams created from that InputStream too then? I'm really not good with file I/O so I have no clue.

Reply:
if you close an inputstream it cascades and closes all its source input streams